### ISSUE TITLE
Do not print `npm ERR!` lines after linting errors

### ISSUE THREAD

**Issue by thany** [eslint/eslint#14201]
**The version of ESLint you are using.**
7.2.0

It might also help to know I'm on Node.js 14.15.3 and NPM 6.14.4.

**The problem you want to solve.**
When there are linting errors after running eslint in the console/terminal, it prints a bunch of `npm ERR!` lines which make it look like eslint had crashed. But in reality it *correctly* returned with an error code. I want those npm error lines gone, because they are not useful, clutter the output, and confuse developers.

Example:
![image](https://user-images.githubusercontent.com/152227/110868387-cbd90680-82c8-11eb-9549-00de0d752654.png)
This is *correctly* an erroneous errorcode, and so it *incorrectly* prints extra information that make it look like eslint crashed, which is confusing and cluttering the console output.

**Your take on the correct solution to problem.**
Maybe there's a way to let NPM know a non-zero errorcode was not actually a problem worthy of the error output shown above.

**Are you willing to submit a pull request to implement this change?**
'fraid not, can't say I'm that much into eslint that I feel able to.

**Comment by mdjermanovic** [comment]
Hi @thany, thanks for the issue! 

This is the behavior of `npm`, I'm not sure if ESLint could control it, and I don't think that ESLint should control it.

You can use the `--silent` flag of the [npm run](https://docs.npmjs.com/cli/v6/commands/npm-run-script) command:

> You can use the --silent flag to prevent showing npm ERR! output on error.

```bash
npm run lint --silent
```

**Comment by thany** [comment]
> I don't think that ESLint should control it

Maybe it's a shortcoming of NPM, in the fact that it cannot distinguish between a script erroring out because something went horribly wrong, and a script sccessfully executing to completion and correctly returning errors.

Running with `--silent` might indeed be a viable solution (or workaround, depending on how you look at it).

**Comment by nzakas** [comment]
As this is the behavior of npm rather than ESLint, I’m closing the issue.
